compiler/rustc_resolve/src/build_reduced_graph.rs

```diff
@@ -258,7 +258,18 @@ impl<'a, 'b> BuildReducedGraphVisitor<'a, 'b> {
                 Ok(ty::Visibility::Restricted(DefId::local(CRATE_DEF_INDEX)))
             }
             ast::VisibilityKind::Inherited => {
-                Ok(ty::Visibility::Restricted(parent_scope.module.normal_ancestor_id))
+                if self.parent_scope.module.kind.is_enum() {
+                    // Anything in an enum has the visibility of the enum.
+                    Ok(*self
+                        .r
+                        .visibilities
+                        .get(&self.parent_scope.module.normal_ancestor_id.expect_local())
+                        .unwrap())
+                } else {
+                    // If it's not in an enum, its visibility is restricted to the trait or module
+                    // that it's defined in.
+                    Ok(ty::Visibility::Restricted(self.parent_scope.module.normal_ancestor_id))
```

Comment:
Should this be:

```suggestion
                    Ok(ty::Visibility::Restricted(self.parent_scope.module.nearest_item_scope().normal_ancestor_id))
```

instead? It seems doing that will restrict trait items' visibility to the module the trait is defined in, whereas not calling it will restrict their visibility to the trait.